core/src/Kokkos_CopyViews.hpp

```diff
@@ -223,8 +223,8 @@ struct ViewFill<ViewType, Layout, ExecSpace, 7, iType> {
       : a(a_), val(val_) {
     Kokkos::parallel_for("Kokkos::ViewFill-7D",
                          policy_type(space, {0, 0, 0, 0, 0, 0},
-                                     {a.extent(0), a.extent(1), a.extent(2),
-                                      a.extent(3), a.extent(5), a.extent(6)}),
+                                     {a.extent(0), a.extent(1), a.extent(3),
+                                      a.extent(4), a.extent(5), a.extent(6)}),
```

Comment:
It stills looks wrong.  We do not have a test for this?

Reply:
The call operator looks like
```C++
 KOKKOS_INLINE_FUNCTION
  void operator()(const iType& i0, const iType& i1, const iType& i3,
                  const iType& i4, const iType& i5, const iType& i6) const {
    for (iType i2 = 0; i2 < iType(a.extent(2)); i2++)
      a(i0, i1, i2, i3, i4, i5, i6) = val;
  };
};
```

Reply:
Yes I saw after I commented.

Reply:
I think adding a test is useful, I'll do it tomorrow.